websites/C/ChompuBot/presence.ts

```diff
@@ -0,0 +1,146 @@
+const presence = new Presence({
+		clientId: "1219713910165209169",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://chompubot.work/bot.png",
+}
+
+const enum Pages {
+	Home = "/",
+	Commands = "/commands",
+	Dashboard = "/dashboard",
+	FAQ = "/faq",
+	Contact = "/contact",
+}
+
+let presenceData: PresenceData;
+
+function isPlayer(): boolean {
+	return !!document.querySelector<HTMLAnchorElement>(
+		"p.text-neutral-200.text-xl.text-center.mb-2"
+	);
+}
+
+presence.on("UpdateData", async () => {
+	const base = document.location.pathname;
+
+	if (/\/dashboard\/guild\/(.*[0-9_].*)\/music-room/gi.test(base)) {
+		let guildName,
+			title,
+			author,
+			playing,
+			pause,
+			timeStartPlayer,
+			timeEndPlayer,
+			StartPlayer,
+			durationPlayer,
+			startTimestamp,
+			endTimestamp;
+
+		if (!isPlayer()) {
+			guildName = document.querySelector<HTMLAnchorElement>(
+				"[data-label='player-requester']"
+			).textContent;
+			title = document.querySelector<HTMLAnchorElement>(
+				"h1.text-large.font-medium.-player-title"
+			);
+			author = document.querySelector<HTMLAnchorElement>(
+				"p.text-small.mt-1.text-foreground\\/80.-player-author"
+			);
+			playing = document.querySelector<HTMLAnchorElement>(
+				"svg.-player-playing"
+			);
+			pause = document.querySelector<HTMLAnchorElement>("svg.-player-pause");
+
+			(timeStartPlayer = document.querySelectorAll<HTMLElement>(
+				"p.text-small.-player-position-start"
+			)[0].textContent),
+				(timeEndPlayer = document.querySelectorAll<HTMLElement>(
+					"p.text-small.text-foreground\\/50.-player-position-end"
+				)[0].textContent),
+				([StartPlayer, durationPlayer] = [
+					presence.timestampFromFormat(timeStartPlayer),
+					(() => {
+						return presence.timestampFromFormat(timeEndPlayer);
+					})(),
+				]),
+				([startTimestamp, endTimestamp] = presence.getTimestamps(
+					StartPlayer,
+					durationPlayer
+				));
```

Comment:
`querySelectorAll(...)[0]` can be simplified for `querySelector(...)`. Also, to make it easier to read, separate these assignments using `;` rather than `,`